pyocd/coresight/funnel.py

```diff
@@ -0,0 +1,75 @@
+# pyOCD debugger
+# Copyright (c) 2023 Protech Engineering
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from .component import CoreSightComponent
+
+LOG = logging.getLogger(__name__)
+
+class TraceFunnel(CoreSightComponent):
+    """@brief CoreSight Trace Funnel"""
+
+    # Register definitions.
+    #
+    # The addresses are offsets from the base address.
+    CSTF = 0x00000000
+    CSTF_ENSX_MASK = 0xFF
+
+    DEVID = 0x00000FC8
+    DEVID_PORTCOUNT_MASK = 0xF
+
+    def __init__(self, ap, cmpid=None, addr=None):
+        """@brief Standard CoreSight component constructor."""
+        super(TraceFunnel, self).__init__(ap, cmpid, addr)
```

Comment:
Although it works fine, you technically don't need the arguments to `super()` since Python 3. Though there are places in pyocd where that hasn't been cleaned up yet. Not a big deal.